### PR TITLE
chore: source controlling cfn templates that will be used for our release process

### PR DESCRIPTION
**Pull request by josecorella**
*Description of changes:*
code_artifact.yml and code_build_parameter_map.json are the same files that live in the [cfn](https://github.com/aws/private-aws-encryption-sdk-java-staging/tree/infra/cfn) branch of the private repos 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: cfn/prod-release.yml [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        - !Ref CryptoToolsKMS
+        - !Ref CodeBuildBatchPolicy
+        - !Ref CodeBuildBasePolicy
+        - !Ref SecretsManagerPolicy
+        - !Ref ParameterStorePolicy
+        - "arn:aws:iam::aws:policy/AWSCodeArtifactReadOnlyAccess"
+        - "arn:aws:iam::aws:policy/AWSCodeArtifactAdminAccess"
+    Metadata:
+      "AWS::CloudFormation::Designer":
+        id: 19b17bdc-5d49-4f12-93e6-6d761b7ce324
```
Comment (texastony) [inline cfn/prod-release.yml:99]: nit: Drop this

File: cfn/prod-release.yml [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                "codebuild:StartBuild",
+                "codebuild:StopBuild",
+                "codebuild:RetryBuild"
+              ]
+            }
+          ]
+        }
+    Metadata:
+      "AWS::CloudFormation::Designer":
+        id: 894a45c5-5dc7-4f11-be65-8bf5e37e289d
```
Comment (texastony) [inline cfn/prod-release.yml:127]: Nit: drop this

File: cfn/prod-release.yml [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+              ],
+              "Resource": [
+                "arn:aws:codebuild:${AWS::Region}:${AWS::AccountId}:report-group/${ProjectName}-*"
+              ]
+            }
+          ]
+        }
+    Metadata:
+      "AWS::CloudFormation::Designer":
+        id: 3dafd088-2792-4a41-b612-d9d049721644
```
Comment (texastony) [inline cfn/prod-release.yml:183]: nit: just do a text search for "AWS::CloudFormation::Designer" and drop these.

They are tags for the icons in the designer tool. They do not encode any information that is relevant to the CFN stack, and they will probably change if this is ever loaded into the designer again.

**Review by texastony — CHANGES_REQUESTED** [review]
Two thoughts:
1. There are no changes to the current infrastructure, correct? This is just documenting it.
2. Remove the metadata fields for cloud designer. 

Cloud Designer is a great tool for validating that the template is good. But it does add these meta tags that are not actually part of the stack.

**Comment by josecorella** [comment]
> Two thoughts:
> 
>     1. There are no changes to the current infrastructure, correct? This is just documenting it.
> 
>     2. Remove the metadata fields for cloud designer.
> 
> 
> Cloud Designer is a great tool for validating that the template is good. But it does add these meta tags that are not actually part of the stack.

1. No changes to the infrastructure. We had talked about adding the webhook settings here in the cfn but that will come in the next iteration of updates to our ci/cd process. Once we are comfortable with pressing the button to run the release we can see about adding the webhook.

**Review by texastony — APPROVED** [review]
Very nice!

**Pull request merged**